### DISCUSSION TITLE
Lesson 6 Error:  contract source info format must be `<path>:<contractname>` or `<contractname>

### DISCUSSION THREAD

**Discussion by essnuked** [Cyfrin/foundry-full-course-cu#1997] [General]
i am trying to deploy my contract with the scripting code and it gives me this error:

'''
estherakamnonu@Esthers-MacBook-Pro foundry-simple-storage-f23 % forge script script/DeploySimpleStorage.s.sol
Error: 
contract source info format must be `<path>:<contractname>` or `<contractname>  
''

'''
[⠒] Unable to resolve imports:
      "../src/SimpleStorage" in "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23/script/DeploySimpleStorge.s.sol"
with remappings:
      forge-std/=/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23/lib/forge-std/src/
[⠆] Compiling 21 files with Solc 0.8.19
[⠰] Solc 0.8.19 finished in 120.49ms
Error: 
Compiler run failed:
Error (6275): Source "src/SimpleStorage" not found: File not found. Searched the following locations: "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23".
ParserError: Source "src/SimpleStorage" not found: File not found. Searched the following locations: "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23".
 --> script/DeploySimpleStorge.s.sol:6:1:
 '''
note: i have this files in my vscode already so i don't know why its not compilling"

**Comment by EngrPips** [comment]
Hello @essnuked, Please show us the command you ran in the terminal to deploy your contract.

**Reply by essnuked** [reply]
forge script script/DeploySimpleStorage.s.sol

**Reply by EngrPips** [reply]
This error `Error (6275): Source "src/SimpleStorage" not found: File not found.` makes me think you don't have `SimpleStorage` contract in your `src` folder. Can you confirm that you have it there?

**Comment by urosognjenovic** [comment]
There's a typo in your file name ```DeploySimpleStorge.s.sol```. Correct it to ```DeploySimpleStorage.s.sol``` and try again.

**Reply by EngrPips** [reply]
Good catch.

**Reply by essnuked** [reply]
i corrected it and ran the command : forge script script/DeploySimpleStorage.s.sol and it gives me this error:
 
Unable to resolve imports:
      "../src/SimpleStorage" in "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23/script/DeploySimpleStorage.s.sol"
with remappings:
      forge-std/=/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23/lib/forge-std/src/
[⠆] Compiling 21 files with Solc 0.8.19
[⠰] Solc 0.8.19 finished in 115.66ms
Error: 
Compiler run failed:
Error (6275): Source "src/SimpleStorage" not found: File not found. Searched the following locations: "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23".
ParserError: Source "src/SimpleStorage" not found: File not found. Searched the following locations: "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23".
 --> script/DeploySimpleStorage.s.sol:6:1:

then when i go to the "problems" area on my erminal,it says this :[{
	"resource": "/Users/estherakamnonu/foundry-f23/foundry-simple-storage-f23/script/DeploySimpleStorage.s.sol",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "6275",
	"severity": 8,
	"message": "Source \"forge-std/Script.sol\" not found: File import callback not supported",
	"startLineNumber": 5,
	"startColumn": 1,
	"endLineNumber": 5,
	"endColumn": 45
}]

**Reply by essnuked** [reply]
but i can see the forge-std file in my folder.

**Reply by EngrPips** [reply]
Maybe you should use native path importing if you are having a problem with remapping.

**Reply by essnuked** [reply]
how do i do that?

**Reply by EngrPips** [reply]
Hello @essnuked, can you try what @Shiva-Sai-ssb suggested? Let's see if that will fix your issue.

**Comment by Shiva-Sai-ssb** [comment]
To debug the error, I tried replacing the import statement in my `DeploySimpleStorage` to:

```solidity
import {SimpleStorage} from "../src/SimpleStorage";
```

I got the same exact error as yours. 

When I reset it to my default import statement:

```solidity
import {SimpleStorage} from "../src/SimpleStorage.sol";
```

The command `forge script script/DeploySimpleStorage.s.sol` ran successfully and gave me the SimpleStorage contract address as the output. Make sure you are importing correctly. If the command didn't work properly, try running the following command:

```sh
forge script script/DeploySimpleStorage.s.sol:DeploySimpleStorage
```

If this doesn’t solve your issue, please push the code to GitHub and share the repo link here so we can help you figure out the issue.

**Comment by wharika18** [comment]
There might be couple of reasons for this :
1) check the path , see that you are actually into the immediate  parent folder 
2) if there are any typo errors in the name of the contract or in the name of the files too , u will face the same error .. for this the following steps will be of help : a) first rectify the typo , then try again the deployment , using the forge script... command , then if it is successful it is  fine, but if you get some error like ```failed to read artifact source file``` then follow the below mentioned process:
 ---> forge clean
---->forge build 
With that most probably you will be able to get through the issue ..
I hope this helps ..

**Comment by EngrPips** [comment]
Thanks for the heads-up @wharika18

**Comment by apuppy** [comment]
I found my mistake is because the script name before(`script/SimpleStorage.s.sol`) is defferent from the course lesson docs(`forge script script/deploySimpleStorage.s.sol --rpc-url=$Sepolia_RPC_URL --private-key=$PRIVATE_KEY --broadcast`).

script/**SimpleStorage.s.sol** VS script/**deploySimpleStorage.s.sol**

- The course lesson docs from Cyfrin Updraft:
[Foundry Fundamentals - Deploying a smart contract on testnet (Sepolia)](https://updraft.cyfrin.io/courses/foundry/foundry-simple-storage/deploying-smart-contract-testnet-sepolia?lesson_format=transcript)

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.

**Comment by mirmohmmadluqman** [comment]
### What you did after it ?
### Did it Resolve ?

**Reply by EngrPips** [reply]
I am unsure what you mean

**Reply by mirmohmmadluqman** [reply]
Did it solved, I am getting error like it
---
![Screenshot from 2025-05-04 19-28-37](https://github.com/user-attachments/assets/30019289-d39b-4d50-aea5-a4e575cf00a7)
![Screenshot from 2025-05-04 19-14-26](https://github.com/user-attachments/assets/357ebf36-34dc-47db-9f1a-e28311c7485e)


- Anvil was running both the times.

**Reply by EngrPips** [reply]
I answered this already in a discussion you opened.
